### PR TITLE
New package: cpputest-3.8

### PR DESCRIPTION
**Pull request by juliendehos**

### CODE REVIEW

File: srcpkgs/cpputest/template [hunk 1]
```diff
@@ -0,0 +1,13 @@
+pkgname="cpputest"
+version="3.8"
+revision=1
+build_style=cmake
+hostmakedepends="cmake"
```
Comment (Vaelatern) [inline srcpkgs/cpputest/template]: redundant

File: srcpkgs/cpputest/template [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+short_desc="CppUTest unit testing and mocking framework for C/C++"
+homepage="http://cpputest.github.io/"
+license="BSD"
```
Comment (Vaelatern) [inline srcpkgs/cpputest/template]: No maintainer?

**Comment by juliendehos** [comment]
Fixed. Thanks.

File: srcpkgs/cpputest/template [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+revision=1
+build_style=gnu-configure
+hostmakedepends="autoconf automake libtool"
+short_desc="CppUTest unit testing and mocking framework for C/C++"
+maintainer="Julien Dehos <dehos@lisic.univ-littoral.fr>"
+homepage="http://cpputest.github.io/"
+license="BSD"
+distfiles="https://github.com/cpputest/${pkgname}/releases/download/v${version}/${pkgname}-${version}.tar.gz"
+checksum="c81dccc5a1bfc7fc6511590c0a61def5f78e3fb19cb8e1f889d8d3395a476456"
+only_for_archs="i686 x86_64 armv6l armv7l armv6l-musl armv7l-musl"
```
Comment (Vaelatern) [inline srcpkgs/cpputest/template]: ..... WHY? This line is probably wrong. Please remove it.

File: srcpkgs/cpputest/template [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+nocross=yes
+do_configure() {
+	cd cpputest_build
+	autoreconf .. -i
+	../configure
+}
```
Comment (Vaelatern) [inline srcpkgs/cpputest/template]: Why do you go down into a dir only to refer to everything above it? Seriously?
Cleaner to
```
pre_configure() {
    autoreconf -fi
}
```
and ignore setting a custom `do_configure`

File: srcpkgs/cpputest/template [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+do_build() {
+	make -C cpputest_build
+}
```
Comment (Vaelatern) [inline srcpkgs/cpputest/template]: Please no. We have a build style for this. If you must set `-C` then do so in `$make_build_args`

File: srcpkgs/cpputest/template [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+do_install() {
+	make -C cpputest_build install
```
Comment (Vaelatern) [inline srcpkgs/cpputest/template]: Again about we have a build style.
`$make_install_args` may be of interest to you.

File: srcpkgs/cpputest/template [hunk 7]
```diff
@@ -0,0 +1,28 @@
+# Template file for 'cpputest'
+pkgname="cpputest"
+version="3.8"
+revision=1
+build_style=gnu-configure
+hostmakedepends="autoconf automake libtool"
```
Comment (Vaelatern) [inline srcpkgs/cpputest/template]: Redundant dependencies.
https://asciinema.org/a/f068kal53a9kn6umrhdduzsqn

File: srcpkgs/cpputest/template [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+short_desc="CppUTest unit testing and mocking framework for C/C++"
+maintainer="Julien Dehos <dehos@lisic.univ-littoral.fr>"
+homepage="http://cpputest.github.io/"
+license="BSD"
+distfiles="https://github.com/cpputest/${pkgname}/releases/download/v${version}/${pkgname}-${version}.tar.gz"
+checksum="c81dccc5a1bfc7fc6511590c0a61def5f78e3fb19cb8e1f889d8d3395a476456"
+only_for_archs="i686 x86_64 armv6l armv7l armv6l-musl armv7l-musl"
+nocross=yes
```
Comment (Vaelatern) [inline srcpkgs/cpputest/template]: Why? Probably wrong. Please remove.

File: srcpkgs/cpputest/template [hunk 9]
```diff
@@ -0,0 +1,13 @@
+pkgname="cpputest"
```
Comment (Vaelatern) [inline srcpkgs/cpputest/template]: This was done, github still can't detect this sort of diff.

File: srcpkgs/cpputest/template (hunk 7 above)
Comment (juliendehos) [inline srcpkgs/cpputest/template]: autoconf removed

File: srcpkgs/cpputest/template (hunk 8 above)
Comment (juliendehos) [inline srcpkgs/cpputest/template]: removed

File: srcpkgs/cpputest/template (hunk 4 above)
Comment (juliendehos) [inline srcpkgs/cpputest/template]: I was following the instructions given with the source code. Fixed.

File: srcpkgs/cpputest/template (hunk 3 above)
Comment (juliendehos) [inline srcpkgs/cpputest/template]: removed but it doesn't compile using musl; I'll try to fix that

**Comment by juliendehos** [comment]
I've finally found a cmake config which seems ok.

**Pull request merged**